src/Features/CSharp/Portable/Completion/KeywordRecommenders/WhereKeywordRecommender.cs

```diff
@@ -109,13 +109,16 @@ private static bool IsTypeParameterConstraintContext(CSharpSyntaxContext context
                 }
             }
 
-            // class C<T> where T : IGoo |
-            // delegate void D<T> where T : IGoo |
-            var constraintClause = token.GetAncestor<TypeParameterConstraintClauseSyntax>();
-
-            if (constraintClause != null)
+            // Do not check constraints right after dot token, as it is probably member access.
+            // See https://github.com/dotnet/roslyn/issues/30785 for example
+            if (!token.IsKind(SyntaxKind.DotToken))
```

Comment:
i think what probably makes more sense is asking if we're after the end token of a *complete* constraint.